cinema/views.py

```diff
@@ -56,3 +88,35 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def filter_queryset(self, queryset):
+        date = self.request.query_params.get("date")
+        movie_id = self.request.query_params.get("movie")
+
+        if date:
+            try:
+                # Match only the date portion of show_time
+                queryset = queryset.filter(show_time__date=date)
+            except ValueError:
+                queryset = queryset.none()
+
+        if movie_id:
+            queryset = queryset.filter(movie_id=movie_id)
+
+        return queryset.distinct()
+
+
+class OrdersViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+
+    def get_queryset(self):
+        return Order.objects.filter(user=self.request.user)
```

Comment:
The `get_queryset` method restricts the orders to those belonging to the authenticated user, ensuring that users can only access their own orders.